src/core/index.ts

```diff
@@ -14,21 +15,41 @@ export class FireboltCore {
   }
 
   async connect(connectionOptions: ConnectionOptions) {
-    const auth = new Authenticator(this.context, connectionOptions);
-    const connection = makeConnection(this.context, connectionOptions);
+    // Create a new httpClient instance for each connection
+    const httpClient =
+      this.options.dependencies?.httpClient || new NodeHttpClient();
+
+    // Create a new context with the new httpClient
+    const connectionContext = {
+      ...this.context,
+      httpClient
+    };
+
+    const auth = new Authenticator(connectionContext, connectionOptions);
+    const connection = makeConnection(connectionContext, connectionOptions);
     await auth.authenticate();
     await connection.resolveEngineEndpoint();
-    const context = {
+    const resourceContext = {
       connection,
-      ...this.context
+      ...connectionContext
     };
-    this.resourceManager = new ResourceManager(context);
+    this.resourceManager = new ResourceManager(resourceContext);
     return connection;
   }
 
   async testConnection(connectionOptions: ConnectionOptions) {
-    const auth = new Authenticator(this.context, connectionOptions);
-    const connection = makeConnection(this.context, connectionOptions);
+    // Create a new httpClient instance for test connection too
```

Comment:
nit: There is a big chunk of code duplicated in this and previous PR now. Let's refactor it into a separate function if possible